app/utils/report/test.py

```diff
@@ -0,0 +1,149 @@
+import os
+
+import models
+import utils
+import utils.db
+import utils.report.common as rcommon
+
+TEST_REPORT_FIELDS = [
+    models.ARCHITECTURE_KEY,
+    models.BOARD_INSTANCE_KEY,
+    models.BOARD_KEY,
+    models.BOOT_ID_KEY,
+    models.BUILD_ID_KEY,
+    models.CREATED_KEY,
+    models.DEFCONFIG_FULL_KEY,
+    models.DEFCONFIG_KEY,
+    models.DEFINITION_URI_KEY,
+    models.DEVICE_TYPE_KEY,
+    models.GIT_BRANCH_KEY,
+    models.GIT_COMMIT_KEY,
+    models.GIT_DESCRIBE_KEY,
+    models.GIT_URL_KEY,
+    models.ID_KEY,
+    models.JOB_ID_KEY,
+    models.JOB_KEY,
+    models.KERNEL_KEY,
+    models.LAB_NAME_KEY,
+    models.METADATA_KEY,
+    models.NAME_KEY,
+    models.STATUS_KEY,
+    models.TEST_CASE_KEY,
+    models.TEST_SET_KEY,
+    models.TIME_KEY,
+    models.VCS_COMMIT_KEY,
+    models.VERSION_KEY,
+]
+
+
+def create_test_report(data, email_format, db_options,
+                       base_path=utils.BASE_PATH):
+    """Create the test suite report email to be sent.
+
+    :param data: The meta-data for the test suite job.
+    :type data: dictionary
+    :param email_format: The email format to send.
+    :type email_format: list
+    :param db_options: The mongodb database connection parameters.
+    :type db_options: dict
+    :param base_path: Path to the top-level storage directory.
+    :type base_path: string
+    :return A tuple with the TXT email body, the HTML email body and the
+    headers as dictionary.  If an error occured, None.
+    """
+    database = utils.db.get_db_connection(db_options)
+
+    job, branch, kernel = (data[k] for k in [
+        models.JOB_KEY,
+        models.GIT_BRANCH_KEY,
+        models.KERNEL_KEY,
+    ])
+
+    specs = {x: data[x] for x in data.keys() if data[x]}
+
+    testsuites = utils.db.find(
+        database[models.TEST_SUITE_COLLECTION],
+        100,
+        0,
+        spec=specs,
+        fields=TEST_REPORT_FIELDS)
+
+    if not testsuites:
+        utils.LOG.warning("Failed to find test suite document")
+        return None
+
+    testsuites = [d for d in testsuites.clone()]
+
+    for ts in testsuites:
+        # ts is a dictionary, where we need to add a new field test_case_list
+        # test_case_list is a list of dictionary with every test case
+        testcase = ts['test_case']
+        ts['test_case_list'] = []
+        for tc in testcase:
+            # tc is a _id from the testcase
+            test_case_doc = utils.db.find_one2(
+               database[models.TEST_CASE_COLLECTION],
+               {"_id": tc})
+            ts['test_case_list'].append(test_case_doc)
+
+        resultlist = [test['status'] for test in ts['test_case_list']]
+        ts['total_tests'] = len(resultlist)
+        ts['total_pass'] = len([e for e in resultlist if e=="PASS"])
+        ts['total_fail'] = len([e for e in resultlist if e=="FAIL"])
+        ts['total_skip'] = len([e for e in resultlist if e=="SKIP"])
+
+    # Remove all the "lava" test suites, they're the boot tests
```

Comment:
It's not quite accurate that the `lava` test suites are the boot tests.  Rather, boot tests only have a `lava` test suite.  Ultimately it would be good to deal with failing boot results for any test plan (say, report that a `usb` test plan failed to complete) as that would also include the plain `boot` test plan without artificially creating a test suite for it.

For the comment here, I think it's better to say that the `lava` test suite doesn't contain any actual results from the defined test plan.